scripts/main.py

```diff
@@ -1,20 +1,65 @@
 #!/usr/bin/env python
 
 import os
+import re
 import codecs
 import sys
 import plugin_loader
+import json
 from jinja2 import Environment, FileSystemLoader
 from jsonschema import validate, exceptions
-from script_helpers import to_json, BASE_DIR
+from script_helpers import (to_json,
+                            BASE_DIR,
+                            REGION_FILE,
+                            extract_filepaths_from_dirs)
 
-REGION_FILE = os.path.join(BASE_DIR, 'region.json')
 REGION_SCHEMA_FILE = os.path.join(BASE_DIR, 'App_Data/regionSchema.json')
 TMPL_FILE = os.path.join(BASE_DIR, 'template_index.html')
 IDX_FILE = os.path.join(BASE_DIR, 'index.html')
 PARTIALS_DIR = os.path.join(BASE_DIR, 'Views/Shared')
 
 
+def prepare_languages():
+    # get app-wide translation files
+    try:
+        language_dir = os.path.join(BASE_DIR, 'locales')
+        plugin_loader.verify_directories_exist([language_dir])
+        app_json_files = [os.path.join(language_dir, file) for file in
+                          os.listdir(language_dir) if file.endswith('.json')]
+    except ValueError as e:
+        msg = 'Failed! Check the app has JSON translation files. {}'.format(e)
+        sys.exit(msg)
+
+    # get plug-ins' translation files
+    plugin_folder_paths = plugin_loader.get_plugin_folder_paths()
+    plugin_locales = [str(path+'/locales') for path in plugin_folder_paths
+                      if os.path.exists(path+'/locales')]
+    plugin_json_files = extract_filepaths_from_dirs(plugin_locales)
+
+    # merge app and plugin translation dicts keyed to language code
+    # prefer app dict translations, if conflict
+    all_json_files = plugin_json_files + app_json_files
+    translations = {}
+
+    for file in all_json_files:
```

Comment:
`file` is a reserved keyword and valid type in python, which is being overwritten for this variable.  Consider renaming this variable to something else.

Reply:
Good catch, thanks!